test/integration/auction.test.js

```diff
@@ -0,0 +1,218 @@
+        .end((err, res) => {
+          if (err) { return done(err) }
+
+          expect(res).to.have.status(200)
+          expect(res.body).to.be.a('object')
+          expect(res.body).to.have.all.keys(...auctionWithIdKeys)
+          expect(res.body.seller).to.be.equal(userData[0].username)
+
+          return done()
+        })
+    })
+
+    it('Should return 404, for Auction not found', function (done) {
+      chai
+        .request(app)
+        .get('/auctions/invalidId')
+        .end((err, res) => {
+          if (err) { return done(err) }
+
+          expect(res).to.have.status(404)
+          expect(res.body).to.be.a('object')
+          expect(res.body.message).to.equal('Auction doesn\'t exist')
+
+          return done()
+        })
+    })
+  })
+
+  describe('POST /auctions', function () {
+    it('Should create a new auction', function (done) {
+      chai
+        .request(app)
+        .post('/auctions')
+        .set('cookie', `${cookieName}=${jwt}`)
+        .send(auctionData)
+        .end((err, res) => {
+          if (err) { return done(err) }
+
+          expect(res).to.have.status(201)
+          expect(res.body).to.be.a('object')
+          expect(res.body.message).to.be.equal('Auction created successfully!')
+
+          return done()
+        })
+    })
+
+    it('Should have enough Item Type in wallet', function (done) {
+      chai
+        .request(app)
+        .post('/auctions')
+        .set('cookie', `${cookieName}=${jwt}`)
+        .send({ ...auctionData, quantity: 5 })
+        .end((err, res) => {
+          if (err) { return done(err) }
+
+          expect(res).to.have.status(403)
+          expect(res.body).to.be.a('object')
+          expect(res.body.message).to.be.equal(`You do not have enough of ${itemType}s!`)
+
+          return done()
+        })
+    })
+
+    it('Should return 401, for Unauthenticated User', function (done) {
+      chai
+        .request(app)
+        .post('/auctions')
+        .end((err, res) => {
+          if (err) { return done(err) }
+
+          expect(res).to.have.status(401)
+          expect(res.body).to.be.a('object')
+          expect(res.body).to.deep.equal({
+            statusCode: 401,
+            error: 'Unauthorized',
+            message: 'Unauthenticated User'
+          })
+
+          return done()
+        })
+    })
+  })
+
+  describe('POST /auctions/bid/:id', function () {
+    it('Should make a new bid with given auctionId', function (done) {
+      chai
+        .request(app)
+        .post(`/auctions/bid/${auctionId}`)
+        .set('cookie', `${cookieName}=${jwt}`)
+        .send({ bid: 500 })
+        .end((err, res) => {
+          if (err) { return done(err) }
+
+          expect(res).to.have.status(201)
+          expect(res.body).to.be.a('object')
+          expect(res.body.message).to.be.equal('Successfully placed bid!')
+
+          return done()
+        })
+    })
+
+    it('Should have a sufficient balance', function (done) {
```

Comment:
`User should have sufficient balance for bidding` would make more sense to me personally. What do you think?